pkg/test/framework/components/istio/config.go

```diff
@@ -161,6 +170,47 @@ func (c *Config) IsMtlsEnabled() bool {
 	return true
 }
 
+func (c *Config) IstioControlPlane() string {
+	data := c.ControlPlaneValues
+	if c.ValuesFile != "" {
+		var err error
+		data, err = file.AsString(filepath.Join(c.ChartDir, c.ValuesFile))
+		if err != nil {
+			return ""
+		}
+	}
+	s, err := image.SettingsFromCommandLine()
+	if err != nil {
+		return ""
+	}
+
+	return fmt.Sprintf(`
+apiVersion: install.istio.io/v1alpha2
+kind: IstioControlPlane
+spec:
+  hub: %s
+  tag: %s
+  values:
+%s
+`, s.Hub, s.Tag, Indent(data, "    "))
+}
+
+// indents a block of text with an indent string
+func Indent(text, indent string) string {
```

Comment:
does this need to be public? If so, seems pretty generic ... maybe move somewhere more common?